### PR TITLE
O3-1712: Analysis: Rework "Expired" in the context of dispensing

### PR DESCRIPTION
**Pull request by mogoodrich**
Note this is dependent on this FHIR PR being approved and merged in:
https://github.com/openmrs/openmrs-module-fhir2/pull/459

### CODE REVIEW

File: src/config-schema.ts
Comment (mogoodrich) [inline src/config-schema.ts:33]: New config parameter to configure the expiration period in days (which is compared against the encounter date)

File: src/medication-request/medication-request.resource.tsx
Comment (mogoodrich) [inline src/medication-request/medication-request.resource.tsx:143]: Updated logic to determine the overall status of Medication Requests linked to an encounter...  this (should) mirror the logic in the encountersWithMedicationRequests named query provided by the FHIR module.

Hopefully, documentation makes sense.

File: src/utils.ts
Comment (mogoodrich) [inline src/utils.ts:96]: Note that we've switched this request and the one below to use the new named query "encountersWithMedicationRequests"

File: src/prescriptions/prescription-details.component.tsx
Comment (mogoodrich) [inline src/prescriptions/prescription-details.component.tsx:36]: updated logic to determine the status of an *individual* medication request

**Comment by mogoodrich** [comment]
Ready for review, see my comment about not merging until the appropriate FHIR PR is approved and merged.

**Review by mseaton — APPROVED** [review]
From a code perspective it seems fine, I'm still not sure conceptually how much I like expiry logic as a frontend-application-level construct.

File: src/config-schema.ts
Comment (pirupius) [reply src/config-schema.ts:33]: Does this mean logic not to allow expired requests has been added as well? And also how does this work in relation to recurring medications?

File: src/medication-request/medication-request.resource.tsx
Comment (pirupius) [inline src/medication-request/medication-request.resource.tsx:36]: 👍🏾

**Review by pirupius — APPROVED** [review]
Just have the one question in regards to how the request expiry related to recurring medications. Has that been put into consideration before being marked as expired? I see the main focus is on the encounter dateTime

File: src/config-schema.ts
Comment (mogoodrich) [reply src/config-schema.ts:33]: Yes, right now you can't dispense orders older than x days old, (which a default of 90)... you are right that this will likely need to be tweaked based on refills, see my comment below.

**Comment by mogoodrich** [comment]
@pirupius regarding expire, yes, it's frustratingly overcomplicated... :)

We are awaiting discussions with our lab teams about how refills fit into the model of the expired, and it seems likely that the logic will need to be tweaks depending on refills, this is just a first past while we figure the out.  If the lab term on your end has any thoughts on this, feel free to have them reach out to Fiona.

Thanks for the review, I'll merge this in after the FHIR PR gets merged (which might not be until next week because I believe Ian is on vacation this week).

**Pull request merged**
